core_lib/structure/soundclip.h

```diff
@@ -41,8 +41,17 @@ class SoundClip : public KeyFrame
     void playFromPosition(int frameNumber, int fps);
     void stop();
 
+    int64_t duration() const;
+    void setDuration(const int64_t &duration);
+
+    void updateLength(int fps);
+
 private:
     std::shared_ptr< SoundPlayer > mPlayer;
+
+    // Duration in seconds. This is stored to update the length of the
+    // frame when the FPS changes.
+    int64_t mDuration;
```

Comment:
Unrelated to the PR, but this reminds me we should really add some actual API docs at some point. @scribblemaniac [experimented with doxygen some time ago](https://github.com/pencil2d/pencil/issues/540#issuecomment-255045764).